Source/Core/Core/PowerPC/JitArm64/JitArm64_Branch.cpp

```diff
@@ -74,6 +76,51 @@ void JitArm64::rfi(UGeckoInstruction inst)
   gpr.Unlock(WA);
 }
 
+void JitArm64::WriteBranchWatch(u32 origin, u32 destination, UGeckoInstruction inst,
+                                BitSet32 gprs_in_use, BitSet32 fprs_in_use)
+{
+  // Use ARM64Reg::X0 as a temporary to avoid moving it later in ABI_CallFunction.
+  static constexpr ARM64Reg branch_watch = ARM64Reg::X0;
+  gpr.Lock(EncodeRegTo32(branch_watch));
+
+  MOVP2R(branch_watch, &m_branch_watch);
+  const ARM64Reg WA = gpr.GetReg();
+  LDRB(IndexType::Unsigned, WA, branch_watch,
+       static_cast<int>(Core::BranchWatch::GetOffsetOfRecordingActive()));
+  FixupBranch branch = TBNZ(WA, true);
+  ABI_PushRegisters(gprs_in_use);
+  m_float_emit.ABI_PushRegisters(fprs_in_use, WA);
+  ABI_CallFunction(
+      m_ppc_state.msr.IR ? &Core::BranchWatch::HitV_fk : &Core::BranchWatch::HitP_fk, branch_watch,
+      Common::BitCast<u64>(Core::FakeBranchWatchCollectionKey{origin, destination}), inst.hex);
+  ABI_PopRegisters(gprs_in_use);
+  m_float_emit.ABI_PushRegisters(fprs_in_use, WA);
```

Comment:
```suggestion
  m_float_emit.ABI_PopRegisters(fprs_in_use, WA);
  ABI_PopRegisters(gprs_in_use);
```

Reply:
Registers must be popped in reverse order of how they were pushed. And you accidentally have a call to PushRegisters here instead of PopRegisters.